crates/paimon/src/spec/types.rs

```diff
@@ -164,118 +112,55 @@ impl FromStr for DataType {
     }
 }
 
-#[allow(dead_code)]
-impl DataType {
-    fn new(is_nullable: bool, type_root: DataTypeRoot) -> Self {
-        Self {
-            is_nullable,
-            type_root,
-        }
-    }
-
-    /// Returns a deep copy of this type with possibly different nullability.
-    /// Impl Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-common/src/main/java/org/apache/paimon/types/DataType.java#L113>
-    fn with_nullable(&self, is_nullable: bool) -> Self {
-        Self {
-            is_nullable,
-            type_root: self.type_root,
-        }
-    }
-
-    /// Returns true if the data type is nullable.
-    ///
-    /// Impl Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-common/src/main/java/org/apache/paimon/types/DataType.java#L59>
-    fn is_nullable(&self) -> bool {
```

Comment:
Is this method should be kept for the `DataType` ?

Reply:
Yes, we can implement in following PRs.

Reply:
OK

Reply:
Tracked at https://github.com/apache/paimon-rust/issues/31